benchmarks/bench_sdcal.py

```diff
@@ -0,0 +1,459 @@
+
+    def time_noff_priority_over_fraction(self):
+        """test_otfraster13 --- check if noff takes priority over fraction."""
+        sdcal(infile=self.infile, outfile=self.outfile, calmode='otfraster', fraction='90%', noff=3)
+
+
+class OTF():
+    """Runtime benchmark tests for task sdcal sky calibration mode = 'otf' : On-The-Fly (OTF) *non-raster*
+    """
+    # Input data
+    infile_squares = 'squares.dec60_cs.ms'
+    infile_lissajous = 'lissajous.ms'
+    infiles = [infile_squares, infile_lissajous]
+    ref_datapath = os.path.join(datapath, 'otf_reference_data')
+    applytable = 'lissajous.edges_new_fraction_0.1.ms_caltable'
+
+    # Output data
+    outfile = 'test_otf.caltable'
+
+    def setup(self):
+        for infile in self.infiles:
+            shutil.copytree(os.path.join(datapath, infile), infile)
+        shutil.copytree(os.path.join(self.ref_datapath, self.applytable), self.applytable)
+
+    def teardown(self):
+        for infile in self.infiles:
+            shutil.rmtree(infile, ignore_errors=True)
+        shutil.rmtree(self.outfile, ignore_errors=True)
+        shutil.rmtree(self.applytable, ignore_errors=True)
+
+    def time_otf01(self):
+        """test_otf01 --- Compute calibration table. calmode='otf' ms=squares.dec60_cs.ms."""
+        sdcal(infile=self.infile_squares, calmode='otf', fraction='10%', noff=-1, width=0.5, elongated=False,
+              applytable='', interp='', spwmap={}, outfile=self.outfile, overwrite=False, field='', spw='')
+
+    def time_otf02(self):
+        """test_otf02 --- Compute calibration table. calmode='otf' ms=squares.dec60_cs.ms edges_fraction=20%."""
+        sdcal(infile=self.infile_squares, calmode='otf', fraction=0.2, noff=-1, width=0.5, elongated=False,
+              applytable='', interp='', spwmap={}, outfile=self.outfile, overwrite=False, field='', spw='')
+
+    def time_otf03(self):
+        """test_otf03 --- Compute calibration table. calmode='otf' ms=lissajous.ms."""
+        sdcal(infile=self.infile_lissajous, calmode='otf', fraction='10%', noff=-1, width=0.5,
+              elongated=False, applytable='',interp='', spwmap={}, outfile=self.outfile, overwrite=False)
+
+    def time_otf04(self):
+        """test_otf04 --- Compute calibration table. calmode='otf' ms=lissajous.ms edges_fraction=20%."""
+        sdcal(infile=self.infile_lissajous, calmode='otf', fraction='20%', noff=-1, width=0.5, elongated=False,
+              applytable='',interp='', spwmap={}, outfile=self.outfile, overwrite=False, field='')
+
+    # To be checked
+    def time_otf05(self):
+        """test_otf05 --- Sky calibration. calmode='otf,apply' ms=squares.dec60_cs.ms."""
+        # In the original test it runs sdcal() twice, first with calmode=apply and then with otf,apply
+        sdcal(infile=self.infile_squares, calmode='otf,apply', fraction='10%', noff=-1, width=0.5,
+              elongated=False, applytable='', interp='', spwmap={}, outfile='', overwrite=False)
```

Comment:
After taking a look again at the original otf cases it looks like there wasn't multiple task executions happening, just a really unfamiliar set up for executing the tasks. I think squares.dec60_cs.ms must be in the test dataset since I'm not seeing new dataset being made/manipulated in the set-up for these test cases.